be/src/storage/rowset/rowset_writer.cpp

```diff
@@ -209,104 +211,161 @@ StatusOr<RowsetSharedPtr> RowsetWriter::build() {
-        ASSIGN_OR_RETURN(auto wfile, _fs->new_writable_file(path));
-
-        // 2. flush segment file
-        auto writer = std::make_unique<SegmentFileWriter>(wfile.get());
-
-        butil::IOBuf segment_data;
-        int64_t remaining_bytes = data.cutn(&segment_data, segment_pb.data_size());
-        if (remaining_bytes != segment_pb.data_size()) {
-            return Status::InternalError(fmt::format("segment {} file size {} not equal attachment size {}",
-                                                     segment_pb.DebugString(), remaining_bytes,
-                                                     segment_pb.data_size()));
-        }
-        while (remaining_bytes > 0) {
-            auto written_bytes = segment_data.cut_into_writer(writer.get(), remaining_bytes);
-            if (written_bytes < 0) {
-                return io::io_error(wfile->filename(), errno);
-            }
-            remaining_bytes -= written_bytes;
-        }
-        if (remaining_bytes != 0) {
-            return Status::InternalError(fmt::format("segment {} write size {} not equal expected size {}",
-                                                     wfile->filename(), segment_pb.data_size() - remaining_bytes,
-                                                     segment_pb.data_size()));
-        }
-        if (config::sync_tablet_meta) {
-            RETURN_IF_ERROR(wfile->sync());
-        }
-        RETURN_IF_ERROR(wfile->close());
+    // 2. flush segment file
+    auto writer = std::make_unique<SegmentFileWriter>(wfile.get());
 
-        if (_context.tablet_schema->keys_type() == KeysType::PRIMARY_KEYS && _context.partial_update_tablet_schema) {
-            auto* partial_rowset_footer = _rowset_txn_meta_pb->add_partial_rowset_footers();
-            partial_rowset_footer->set_position(segment_pb.partial_footer_position());
-            partial_rowset_footer->set_size(segment_pb.partial_footer_size());
+    butil::IOBuf segment_data;
+    int64_t remaining_bytes = data.cutn(&segment_data, segment_pb.data_size());
+    if (remaining_bytes != segment_pb.data_size()) {
+        return Status::InternalError(fmt::format("segment {} file size {} not equal attachment size {}",
+                                                 segment_pb.DebugString(), remaining_bytes, segment_pb.data_size()));
+    }
+    while (remaining_bytes > 0) {
+        auto written_bytes = segment_data.cut_into_writer(writer.get(), remaining_bytes);
+        if (written_bytes < 0) {
+            return io::io_error(wfile->filename(), errno);
         }
+        remaining_bytes -= written_bytes;
+    }
+    if (remaining_bytes != 0) {
+        return Status::InternalError(fmt::format("segment {} write size {} not equal expected size {}",
+                                                 wfile->filename(), segment_pb.data_size() - remaining_bytes,
+                                                 segment_pb.data_size()));
+    }
+    if (config::sync_tablet_meta) {
```

Comment:
Can we flush only once for one ingestion?

Reply:
This `sync` will be called only once for each segment. It is just ok?

Reply:
OK